src/string.rs

```diff
@@ -151,3 +151,21 @@ impl Deref for String {
     unsafe { &*(self as *const _ as *const Value) }
   }
 }
+
+impl Into<String> for Value {
+  fn into(self) -> String {
+    // TODO: do this check in debug mode only
+    if !self.is_string() {
+      panic!("unable to convert")
+    } 
+    
+    unsafe { std::mem::transmute(self) }
+  }
+}
```

Comment:
```c++
void v8::Function::CheckCast(Value* that) {
  i::Handle<i::Object> obj = Utils::OpenHandle(that);
  Utils::ApiCheck(obj->IsCallable(), "v8::Function::Cast",
                  "Could not convert to function");
}

void v8::Boolean::CheckCast(v8::Value* that) {
  i::Handle<i::Object> obj = Utils::OpenHandle(that);
  Utils::ApiCheck(obj->IsBoolean(), "v8::Boolean::Cast",
                  "Could not convert to boolean");
}

void v8::Name::CheckCast(v8::Value* that) {
  i::Handle<i::Object> obj = Utils::OpenHandle(that);
  Utils::ApiCheck(obj->IsName(), "v8::Name::Cast", "Could not convert to name");
}
```

Lot of `CheckCast` methods use types from `v8::internal` namespace which I'm pretty sure we don't want to port...